tuf/client/updater.py

```diff
@@ -1217,7 +1217,7 @@ def _check_hashes(self, file_object, trusted_hashes):
 
 
 
-  def _hard_check_file_length(self, file_object, trusted_file_length):
+  def _check_file_length(self, file_object, trusted_file_length):
```

Comment:
Now that you made me look at this function, I'll comment on it (this may be a bit unrelated to your PR): 

It does 
```
file_object.seek(0)
observed_length = len(file_object.read())
```
which seems wildly unoptimal. I imagine this would be better:
```
file_object.seek(0,2)
observer_length = file_object.tell()
```

Reply:
Looks good to me too!  Please add a comment to explain "2" if you add this as it's a magic number here...

Reply:
Thanks for the suggestion @jku, I pushed an additional patch to implement this optimisation with you listed as co-author.